src/azure/storage/credential.rs

```diff
@@ -16,5 +16,36 @@ pub enum Credential {
     /// associated with the subscription that contains the storage account.
     ///
     /// ref: <https://docs.microsoft.com/rest/api/storageservices/authorize-with-azure-active-directory>
-    BearerToken(String),
+    BearerToken(String, String),
```

Comment:
It's better to store a `DateTime` directly. `is_valid` will check it every time we call `load_credential`.